internal/fourslash/fourslash.go

```diff
@@ -544,17 +549,14 @@ func (f *FourslashTest) verifyCompletionsWorker(t *testing.T, expected *Completi
 		TextDocumentPositionParams: f.currentTextDocumentPositionParams(),
 		Context:                    &lsproto.CompletionContext{},
 	}
-	resMsg := f.sendRequest(t, lsproto.MethodTextDocumentCompletion, params)
+	resMsg, result, resultOk := sendRequest(t, f, lsproto.TextDocumentCompletionInfo, params)
 	if resMsg == nil {
 		t.Fatalf(prefix+"Nil response received for completion request", f.lastKnownMarkerName)
 	}
-	result := resMsg.AsResponse().Result
-	switch result := result.(type) {
-	case *lsproto.CompletionList:
-		f.verifyCompletionsResult(t, f.currentCaretPosition, result, expected, prefix)
-	default:
-		t.Fatalf(prefix+"Unexpected response type for completion request: %v", result)
+	if !resultOk {
+		t.Fatalf(prefix+"Unexpected response type for completion request: %T", resMsg.AsResponse().Result)
 	}
```

Comment:
It'd be ideal if `sendRequest` (or a wrapper function) automatically errored based on `resMsg` and `resultOk` and used the passed `Method` for the errors.

Reply:
I wanted to do this, but it means I have to also pass in `prefix` which felt strange...